XenAdmin/Core/Updates.cs

```diff
@@ -416,15 +416,21 @@ private static bool IsSuitableForXenCenterAlert(XenCenterVersion toUse, Version
                 if (master == null || pool == null)
                     continue;
 
-                var serverVersions = GetServerVersions(master, xenServerVersions);
+                var serverVersions = new List<XenServerVersion>();
+                foreach (Host host in hosts)
+                {
+                    var serverVersion = GetServerVersions(host, xenServerVersions);
+                    serverVersions.AddRange(serverVersion);
+                }
+                serverVersions = serverVersions.Distinct().ToList();
 
                 if (serverVersions.Count == 0)
                     continue;
 
                 foreach (XenServerVersion xenServerVersion in serverVersions)
                 {
-                    XenServerVersion version = xenServerVersion;
-                    List<XenServerPatch> patches = xenServerPatches.FindAll(patch => version.Patches.Contains(patch));
+                    XenServerVersion patchApplicableVersion = xenServerVersion;

```

Comment:
When I suggested the rename of the "version" parameter I was referring to the definition of the `PatchCanBeInstalledOnHost` method, not this local variable.